packages/vscode-extension/src/qm/vsc_ui.ts

```diff
@@ -446,7 +448,85 @@ export class VsCodeUI implements UserInteraction {
   }
 
   async selectFolder(config: SelectFolderConfig): Promise<Result<SelectFolderResult, FxError>> {
-    return this.selectFileInQuickPick(config, "folder", config.default);
+    if (!TreatmentVariableValue.useFolderSelection) {
+      // control group
+      return this.selectFileInQuickPick(config, "folder", config.default);
+    } else {
+      // treatment group
+      const disposables: Disposable[] = [];
+      try {
+        const quickPick = window.createQuickPick<FxQuickPickItem>();
+        quickPick.title = config.title;
+        if (config.step && config.step > 1) {
+          quickPick.buttons = [QuickInputButtons.Back];
+        }
+        quickPick.placeholder = config.placeholder;
+        quickPick.ignoreFocusOut = true;
+        quickPick.matchOnDescription = true;
+        quickPick.matchOnDetail = true;
+        quickPick.canSelectMany = false;
+        return await new Promise<Result<SelectFolderResult, FxError>>(
+          async (resolve): Promise<void> => {
+            // set options
+            quickPick.items = [
+              {
+                id: "default",
+                label: localize("teamstoolkit.qm.defaultFolder"),
+                description: config.default,
+              },
+              {
+                id: "browse",
+                label: `$(folder) ${localize("teamstoolkit.qm.browse")}`,
+              },
+            ];
+
+            const onDidAccept = async () => {
+              const selectedItems = quickPick.selectedItems;
+              if (selectedItems && selectedItems.length > 0) {
+                const item = selectedItems[0];
+                ExtTelemetry.sendTelemetryEvent(TelemetryEvent.SelectFolder, {
```

Comment:
Do we need to add this event since question model already sends events with question name and answer?

Reply:
The answer is user's folder path, and we cannot query it in telemetry data. This telemetry event helps us to understand which option they choose.